src/CMakeLists.txt

```diff
@@ -0,0 +1,148 @@
+get_filename_component(PROJECT_ROOT "${CMAKE_CURRENT_LIST_DIR}/.." ABSOLUTE)
+
+# Read version from file
+file(READ "${PROJECT_ROOT}/VERSION" VERSION_STR)
+string(REGEX MATCH "^([0-9]+\\.[0-9]+\\.[0-9]+)(.*)" VERSION_STR "${VERSION_STR}")
```

Comment:
Is the regex guaranteed to succeed?
Maybe
```cmake
if(NOT VERSION_STR)
    message(FATAL_ERROR "Failed to extract version from ${PROJECT_ROOT}/VERSION")
endif()
```

Or just read the file using `file(READ)` + `string(STRIP)`?

Reply:
as it appears, in case of broken VERSION file the configuration would still fail on later steps (project() command would fail) but an explicit check at least makes the the error message more readable